stock_picking_invoice_link/models/sale_order.py

```diff
@@ -3,23 +3,32 @@
 # License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
 
 from odoo import models
-from odoo.tools import float_compare
+from odoo.tools import float_compare, float_is_zero
 
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
     def get_stock_moves_link_invoice(self):
-        return self.mapped('move_ids').filtered(
-            lambda x: (
-                x.state == 'done' and not (any(
-                    inv.state != 'cancel' for inv in x.invoice_line_ids.mapped(
-                        'invoice_id'))) and not x.scrapped and (
-                    x.location_dest_id.usage == 'customer' or
-                    (x.location_id.usage == 'customer' and
-                     x.to_refund))
-            )
-        )
+        moves_linked = self.env["stock.move"]
+        for stock_move in self.move_ids:
+            if stock_move.state != 'done' or stock_move.scrapped or (
```

Comment:
@sergio-teruel IMHO a helper function or field to filter would be welcome in order to better readability and comprehension (apart of the fact it could be overidden easily too)